db/migrate/20210907022247_make_post_id_optional_bookmarks.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class MakePostIdOptionalBookmarks < ActiveRecord::Migration[6.1]
+  def up
+    remove_index :bookmarks, [:user_id, :post_id], unique: true
+    change_column_null :bookmarks, :post_id, true
+    add_index :bookmarks, [:user_id, :post_id, :topic_id], unique: true
+  end
+
+  def down
+    Bookmark.where(post_id: -1).delete_all
+    Bookmark.where(post_id: nil).delete_all
```

Comment:
I think we can do this in SQL and in a single query.

Reply:
Ugh yeah completely forgot the no model stuff, was meant to go back and clean this up, my bad